transformers/tests/utils.py

```diff
@@ -58,8 +58,7 @@ def require_tf(test_case):
 
 
 if _torch_available:
-    import torch
-    # Set the CUDA_VISIBLE_DEVICES environment variable to select a GPU.
-    torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+    # Set the USE_CUDA environment variable to select a GPU.
+    torch_device = "cuda" if os.environ.get("USE_CUDA") else "cpu"
```

Comment:
I'd rather use an explicit env var, that way we can use the same for TF when applicable (not sure if TF supports CUDA_VISIBLE_DEVICES)

I think i'd rather have GPU tests be opt-in for now (even on GPU machine)

Reply:
(PS: I don't have a *super* strong opinion on this so other people, feel free to chime in.)

@LysandreJik @thomwolf

Reply:
TF supports `CUDA_VISIBLE_DEVICES`.

I did it that way because I believe that most people would want to run tests on the GPU if they have a CUDA-compatible GPU (unless there's a drawback that I missed).

I also considered adding an environment variable to override the default behavior. Asking to run the tests on the GPU when there's no GPU available would be an error. Would that answer your concerns?

Reply:
I'm adding a commit that implements the above.

Just evaluating `os.environ.get("USE_CUDA")` in a boolean context isn't ideal because `USE_CUDA=0`, `USE_CUDA=no`, `USE_CUDA=false` will all be interpreted as "yes, use CUDA". This is a very common usability issue. Hence the `get_bool_from_env_var` function.

The [TF docs](https://www.tensorflow.org/api_docs/python/tf/test/is_gpu_available) mentions alternatives to CUDA. To be future proof, I chose a generic name `USE_GPU` rather than the more specific `USE_CUDA`. You get the last word on this.

Reply:
@thomwolf @LysandreJik feel free to chime in on this point